EDP-380C-28-R2-Lab3.R

```diff
@@ -12,3 +12,500 @@ source("scripts/rmvnorm.R")
+    
+    dat <- cbind(Y, X)
+    return(dat)
+}
+
+test1 <- method1Fun(n, x_p, y_p, betas, R2)
+method1Mod <- myLM(test1[,1], test1[,2:7])
+#           Estimate          SE    t value p value
+# b0    25.0182701 0.012690524 1971.41356       0
+# b1     0.9843795 0.013048351   75.44091       0
+# b2     1.0166090 0.009238173  110.04438       0
+# b3     0.9901229 0.007552943  131.09100       0
+# b4     0.9876760 0.006538371  151.05843       0
+# b5     1.0042164 0.005881917  170.72944       0
+# SD(e)  4.0129881          NA         NA      NA
+# R2     0.5893671          NA         NA      NA
+
+# get varE parameter to compare to estimate
+params <- transFun(x_p, y_p)
+# generate the covariance matrix, Sigma
+covarMatrix <- genSigma(params)
+# generate the Xs
+X <- rmvnorm(n, x_p$mus, covarMatrix)
+# add intercept
+X <- cbind(matrix(1, NROW(X)), X)
+# determine the variance of Y given betas and covarMatrix
+varY <- t(betas) %*% covarMatrix %*% betas # eq 22
+# determine the error variance 
+varE <- varY * (1/R2 - 1) 
+# generate y
+# expectation of Y
+Ey <- params$mus[1] - t(params$mus[-1]) %*% betas
+
+pop <- c(Ey, betas, sqrt(varE), R2)
+estimate <- method1Mod[,1]
+difference <- pop - estimate
+
+difTab <- cbind(pop, estimate, difference)
+#           pop   estimate   difference
+# b0    25.000000 25.0182701 -0.018270136
+# b1     1.000000  0.9843795  0.015620494
+# b2     1.000000  1.0166090 -0.016608976
+# b3     1.000000  0.9901229  0.009877059
+# b4     1.000000  0.9876760  0.012324021
+# b5     1.000000  1.0042164 -0.004216436
+# SD(e)  4.825922  4.0129881  0.812933981
+# R2     0.500000  0.5893671 -0.089367060
+
+# Method 2, generate from marginal correlations
+# Test the Method 1 function using five predictors with n = 100000 that are all cor- related 0.15 with variances from 1 to 5. The correlations with Y ought to be −.15, −.50, .15, 0.30, and 0.20. The mean and variance of Y are μy = 10, σy = 4. Set the seed to 1237
+set.seed(1237)
+n <- 100000
+
+x_p <- list(
+    mus = c(0,0,0,0,0),
+    sigmas = c(sqrt(1), sqrt(2), sqrt(3), sqrt(4), sqrt(5)),
+    rhos = rep(.15, 10)
+)
+
+y_p <- list(
+    mus = c(10),
+    sigmas = c(4),
+    rhos = c(-.15, -.5, .15, .30, .20)
+)
+
+method2Fun <- function(n, x_p, y_p, betas){
+    #get params into workable form
+    params <- transFun(x_p, y_p)
+    #gereate covrmtrix for x
+    covarMatrix <- genSigma(x_p)
+    # generate X
+    X <- rmvnorm(n, x_p$mus, covarMatrix)
+    #intercept
+    X <- cbind(matrix(1, NROW(X)), X)
+    # generate the covariance matrix, Sigma
+    covarMatrix <- genSigma(params)
+    # solve for the betas
+    betas <- solve(covarMatrix[-1, -1]) %*% covarMatrix[1, -1]
+    # determine the error variance 
+    varE <- covarMatrix[1,1] - t(betas) %*%  covarMatrix[1, -1]
+    # generate y
+    # expectation of Y
+    Ey <- params$mus[1] - t(params$mus[-1]) %*% betas
+    # generate y
+    Y <- Ey[1] + X[,-1] %*% betas + rnorm(n, 0, sqrt(sigmaE))
```

Comment:
Similar to the above comment, you are not creating `sigmaE` within this function. You also add an intercept and then remove it.